res/notes/modular-auth.md

```diff
@@ -0,0 +1,131 @@
+## Modular Auth notes
+
+- Authentication service
+    - Each service will have it's own repository to hold implementation
+    
+    - Things to consider
+        - ui
+            - Forms or any view component should be defined for each component.
+            - Idealy accounts/client will passed on the ui config method and it will then be used by the ui components 
+        - data store
+            - Store auth details for document based stores
+                - How indexes are handled?
+            - map to relational entries (for sql implmentation)
+        - added endpoints
+            - should expose the express/koa/hapi middleware if needed.
+    
+    - Register service plugins
+        - Register on client
+        - Register on server
+        - Suggested API
+        ```typescript
+        // client code
+        const accountsClient = new AccountsClient({
+            authServices: {
```

Comment:
It's possible but most services (especially oauth) will need some configurations on the client side as well so it's probably easier to the let the developer configure it's own instance and supply it to the `accountsClient`.
Also I think it will easier for developers to write client with other programming languages that use an AccountsServer service.